tsup.config.ts

```diff
@@ -0,0 +1,13 @@
+import { defineConfig } from "tsup";
+
+export default defineConfig({
+  outDir: "dist/lib",
+  entry: ["./lib/index.ts"],
+  format: ["cjs", "esm"],
+  target: "node18",
+  sourcemap: true,
+  clean: true,
+  minify: false,
+  shims: true,
+  dts: true,
+});
```

Comment:
This can be moved into `package.json` if desired, but when in a typescript/javascript file, you get type hints from your editor, which is why I put it in its own config.